opentracing-api-extensions-tracer/src/test/java/io/opentracing/contrib/api/tracer/APIExtensionsSpanTest.java

```diff
@@ -207,6 +208,38 @@ public void testOnSetTagBoolean() throws InterruptedException {
         }
     }
 
+    @Test
+    public void testOnSetTagNullKey() throws InterruptedException {
+        Span testSpan = spanFactory.create();
+        APIExtensionsSpan span = new APIExtensionsSpan(testSpan, null,
+                0, 0, new ConcurrentHashMap<String, Object>());
+        SpanObserver observer = Mockito.mock(SpanObserver.class);
+        span.addSpanObserver(observer);
+
+        span.setTag((String)null, "testvalue");
+
+        verify(observer).onSetTag(span, null, "testvalue");
+        if (testSpan != null) {
```

Comment:
Is this null check necessary?

Reply:
Yes, because the tests are run twice, once with a null span.